### PR TITLE
Feature/wa 284 dependency injection

### PR DESCRIPTION
**Pull request by DmitryBespalov**
- Added ApplicationServiceRegistry
- Replaced calls to AuthenticationApplicationService with registry calls

Next steps (in future PRs):

- [ ] Add DomainRegistry for domain model layer
- [ ] Refactor tests to depend on AuthenticationService, not on the account.
- [ ] Add tests for Authentication Service
- [ ] Move domain model layer source code to separate module
- [ ] Move application layer code to separate module
- [ ] Move "ServiceLayer" code to PortAdapters module

### CODE REVIEW

File: safe/safe/AppFlow/AppFlowCoordinatorTests.swift [hunk 1]
```diff
@@ -5,15 +5,9 @@
 import XCTest
 @testable import safe
 
-class AppFlowCoordinatorTests: XCTestCase {
+class AppFlowCoordinatorTests: AbstractAppTestCase {
```
Comment (sche) [inline safe/safe/AppFlow/AppFlowCoordinatorTests.swift:8]: Should we change file template as well?

File: safe/safe/ApplicationLayer/ApplicationServiceRegistry.swift [hunk 2]
```diff
@@ -0,0 +1,13 @@
+//
+//  Copyright © 2018 Gnosis. All rights reserved.
+//
+
+import Foundation
+
+class ApplicationServiceRegistry: AbstractRegistry {
+
+    class func authenticationService() -> AuthenticationApplicationService {
```
Comment (sche) [inline safe/safe/ApplicationLayer/ApplicationServiceRegistry.swift]: how about class var here?

**Review by sche — APPROVED** [review]

File: safe/safe/ApplicationLayer/ApplicationServiceRegistry.swift (hunk 2 above)
Comment (DmitryBespalov) [reply safe/safe/ApplicationLayer/ApplicationServiceRegistry.swift]: ok, looks more swifty

File: safe/safe/AppFlow/AppFlowCoordinatorTests.swift (hunk 1 above)
Comment (DmitryBespalov) [reply safe/safe/AppFlow/AppFlowCoordinatorTests.swift:8]: mm.. not really, I think there're changes in next prs that affect the tests.

**Pull request merged**